### ISSUE TITLE
Fix tree select

### ISSUE THREAD

**Issue by cormacmccarthy** [ExactTarget/fuelux#1724]
If an API method is an action, that stated action should consistently take place when called. However, for tree, this is not the case. `selectItem` both selects _and deselects_. This is not only confusing but potentially could cause errors and bugs in implementor's code, because they will need to check the item before calling select on it to make sure it is not selected in order to guarantee that their selection method doesn't undo the very thing they are trying to do. The method should take care of that for them.

* calling `.tree('selectItem'` ought to ensure the item is selected, even if it was previously
* calling `.tree('deselectItem'` ought to ensure the item is _not_ selected, even if it previously wasn't
* calling `.tree('toggleSelect'` ought to behave as `selectItem` currently behaves, in that it should merely toggle the selection state on the item passed in.

related to #1715

**Comment by mkhahani** [comment]
Correct. The same goes for click event. The click should always select the node even if it is selected before. I think no one clicks the node to deselect it! I also see this behavior in **Repeater**.

**Comment by jjuu** [comment]
Year, that confuses me a lot. What a waste of time.
